### PR TITLE
feat: added forgot-password page

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Added Forgot Password and Reset Password Page

## Description

 - Added handlers for forgot password and reset password
 - Used EmailJS for sending mail containing reset password link
 - Added Forgot Password, Reset Password and MailSent components
 
 Deploy preview - [https://echo-store-ecommerce-14xovr5y8-srv-pranjal.vercel.app/](https://echo-store-ecommerce-14xovr5y8-srv-pranjal.vercel.app/)
 
Animation

### CODE REVIEW

**Review by bharati-21 — APPROVED** [review]
Appreciation for well wriiten and understandable code.
Also, adding forgot password feature was new for me and I got to learn something new. Thank you. 

Looks good to be merged.

File: src/pages/Auth/ForgotPassword.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import { showToast } from "utilities";
+import axios from "axios";
+import { useDocumentTitle } from "hooks";
+import { ThreeDots } from "react-loader-spinner";
```
Comment (bharati-21) [inline src/pages/Auth/ForgotPassword.js:5]: Appreciation - great to see you are trying out something new!

File: src/backend/controllers/AuthController.js [hunk 2]
```diff
@@ -81,11 +83,90 @@ export const loginHandler = function (schema, request) {
...
+
+    const token = sign(
+      { _id: foundUser._id, email },
+      process.env.REACT_APP_JWT_SECRET,
+      { expiresIn: 60 * 60 * 15 }
+    );
+    const link = `${protocol}//${host}/reset-password/:${token}`;
+
+    try {
+      let status = await emailjs.send(
```
Comment (bharati-21) [inline src/backend/controllers/AuthController.js:122]: Well written. Again, got to learn something new. Thank you.

**Review by akash-1318 — COMMENTED** [review]
First of fall great work with the forgot password really understandable and well written code with meaningful naming , no changes require from my end . Great work .

File: src/pages/Auth/ForgotPassword.js (hunk 1 above)
Comment (srv-pranjal) [reply src/pages/Auth/ForgotPassword.js:5]: Thanks a lot!

**Pull request merged**